_posts/2020-12-16-how-to-update-to-terraform-13.adoc

```diff
@@ -0,0 +1,233 @@
+---
+title: How to update to Terraform 13
+categories: Upgrades
+image: /assets/img/guides/refresh_icon.png
+excerpt: Learn how to update the Gruntwork Reference Architecture and your infrastructure code that depends on the Gruntwork Infrastructure as Code Library to Terraform 0.13.
+tags: ["aws", "terraform", "terragrunt"]
+cloud: ["aws"]
+redirect_from: /static/guides/upgrades/how-to-update-to-terraform-13/
+---
+:page-type: guide
+:page-layout: post
+
+:toc:
+:toc-placement!:
+
+// GitHub specific settings. See https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74 for details.
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+toc::[]
+endif::[]
+
+== Intro
+
+This guide will walk you through how to update the https://gruntwork.io/reference-architecture/[Gruntwork Reference
+Architecture] and any code that depends on the
+https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library] to
+https://www.terraform.io/upgrade-guides/0-13.html[Terraform 0.13]. Terraform 0.13 introduces a number of new features
+and fixes, but it also has a number of backwards incompatibilities that have to be incorporated into your codebase.
+
+=== What you'll learn in this guide
+
+This guide consists of three main sections:
+
+<<core_concepts>>::
+  An overview of Terraform 0.13 and why it is important to update your code for compatibility.
+
+<<deployment_walkthrough>>::
+  The steps you need to take to update your code relying on the Gruntwork Infrastructure as Code library and your
+  version of the Gruntwork Reference Architecture to work with Terraform 0.13. Includes a
+  <<compatibility_table,version compatibility table>> you can use as a reference to know which Gruntwork Repo version
+  tag is compatible with Terraform 0.13.
+
+
+[[core_concepts]]
+== Core Concepts
+
+=== Background
+
+https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-13[Terraform 0.13 was released on August 10th, 2020].
+Some of the major new features in 0.13 include:
+
+. The ability to use `count`, `for_each`, and `depends_on` with modules.
+. A new `required_providers` block that makes it much easier to work with third-party providers.
+. Custom variable validation.
+
+For more info, check out the https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-13[announcement blog post].
+
+0.13 is a major new release for Terraform, which means it includes a number of backwards incompatible changes. We have
```

Comment:
NIT: I understand the context here as "They've made loads of changes, so this is a major few changes you have to catch up with", but it might be worth avoiding any confusion this could lead to - e.g. "major" next to the version number might make people think twice what versioning system we're following. 

A few suggestions that could fit in well there if you decide to take upon this comment: like "important", "big", "significant", "considerable".

Reply:
Hm, I think we _do_ want to refer to "major" as in the versioning system. Terraform is pre 1.0, so bumping the "minor" version number in `MAJOR.MINOR.PATCH`, when `MAJOR` is still 0, _is_ considered a "major" release.

Reply:
Ahh my bad in this case and feel free to ignore me! Thanks for letting me know - I was following the semver guide we've got, but it doesn't explicitly mention this, so I did wonder.

Reply:
Roger!